packages/common/http/src/transfer_cache.ts

```
@@ -335,3 +382,9 @@ function appendMissingHeadersDetection(
     },
   });
 }
+
+function mapRequestUrl(url: string, originMap: Record<string, string>): string {
+  const origin = new URL(url, 'resolve://').origin;
+  const mappedOrigin = originMap[origin];
+  return mappedOrigin ? url.replace(origin, mappedOrigin) : url;
```

I'm curious if we should make some dev mode checks (`if (ngDevMode) { ... }`) to ensure that the provided configuration contains origins, but not URLs with extra paths, e.g. avoid situation like this:

```
{
  'http://internal-domain.com:8080/some/extra/paths': 'https://external-domain.com'
}
```

@alan-agius4 WDYT?

---

Agree.

---

Are you thinking of a warning or a runtime error here?

---

> Would it be enough to check if new URL(origin).pathname === '/' or do you have something else in mind?

Yes, that should be sufficient 👍 

> Are you thinking of a warning or a runtime error here?

We should throw an error in this case, since it'd be an unsupported configuration.

---

I added the validation only in dev mode. Let me know if you're ok with the implementation.

---

@jsaguet thanks for adding the check. I think we can simplify the logic a bit (and also make it less excessive) - in this code:

```
  const origin = new URL(url, 'resolve://').origin;
  const mappedOrigin = originMap[origin];
  return mappedOrigin ? url.replace(origin, mappedOrigin) : url;
```

We get an origin for the current URL already, so we don't need to do more checks. We should only verify that the `mappedOrigin` is valid, i.e.:

```
  const origin = new URL(url, 'resolve://').origin;
  const mappedOrigin = originMap[origin];
  if (mappedOrigin) {
    if (ngDevMode) {
      verifyMappedOrigin(mappedOrigin);
    }
    return url.replace(origin, mappedOrigin);
  }
  return url;
```

---

Sorry, I misunderstood your first comment. 
I updated the implementation so we're now actually checking the mapped origin only and ignoring potential additional paths set to the "internal" origin.